internal/pkg/cli/task_run.go

```diff
@@ -719,12 +719,16 @@ func (o *runTaskOpts) Execute() error {
 
 	// NOTE: if image is not provided, then we build the image and push to ECR repo
 	if o.image == "" {
-		uri, err := o.repository.Login()
+		uri, err := o.repository.URI()
+		if err != nil {
+			return fmt.Errorf("get ECR repository URI: %w", err)
+		}
```

Comment:
nit: we can maybe move ethis line to just before `o.image = fmt.Sprintf(fmtImageURI, uri, tag)`, if that is the only place we need `uri` for.